src/Features/Core/Portable/Completion/Providers/AbstractTypeImportCompletionProvider.cs

```diff
@@ -0,0 +1,161 @@
+            {
+                return;
+            }
+
+            var syntaxContext = await CreateContextAsync(document, completionContext.Position, cancellationToken).ConfigureAwait(false);
+            await AddCompletionItemsAsync(document, syntaxContext, completionContext, cancellationToken).ConfigureAwait(false);
+        }
+
+        private async Task AddCompletionItemsAsync(Document document, SyntaxContext context, CompletionContext completionContext, CancellationToken cancellationToken)
+        {
+            if (!context.IsTypeContext)
+            {
+                return;
+            }
+
+            using (Logger.LogBlock(FunctionId.Completion_TypeImportCompletionProvider_GetCompletionItemsAsync, cancellationToken))
+            {
+                var node = context.LeftToken.Parent;
+                var project = document.Project;
+
+                // Find all namespaces in scope at current cursor location, 
+                // which will be used to filter so the provider only returns out-of-scope types.
+                var namespacesInScope = GetNamespacesInScope(node, context.SemanticModel, cancellationToken);
+                Action<CompletionItem> handleAccessibleItem = item => AddItems(item, completionContext, namespacesInScope);
+
+                await _typeImportCompletionService.GetAccessibleTopLevelTypesFromProjectAsync(project, handleAccessibleItem, cancellationToken)
+                    .ConfigureAwait(false);
+
+                var compilation = await project.GetCompilationAsync(cancellationToken).ConfigureAwait(false);
+                foreach (var reference in compilation.References)
+                {
+                    cancellationToken.ThrowIfCancellationRequested();
+
+                    var declarationsInReference = ImmutableArray<CompletionItem>.Empty;
+                    if (reference is CompilationReference compilationReference)
+                    {
+                        await _typeImportCompletionService.GetAccessibleTopLevelTypesFromCompilationReferenceAsync(
+                            project.Solution,
+                            compilation,
+                            compilationReference,
+                            handleAccessibleItem,
+                            cancellationToken).ConfigureAwait(false);
+                    }
+                    else if (reference is PortableExecutableReference peReference)
+                    {
+                        _typeImportCompletionService.GetAccessibleTopLevelTypesFromPEReference(
+                            project.Solution,
+                            compilation,
+                            peReference,
+                            handleAccessibleItem,
+                            cancellationToken);
+                    }
+                }
+            }
+
+            static void AddItems(CompletionItem item, CompletionContext completionContext, ImmutableHashSet<string> namespacesInScope)
```

Comment:
add an explicit 'return' statement above this.  it helps make it clear that no extra code for the main control flow will come after the local function.